src/EntityManager.php

```diff
@@ -406,13 +409,13 @@ public function getReference(string $entityName, mixed $id): object|null
     public function clear(): void
     {
         $this->unitOfWork->clear();
+
+        $this->unitOfWork = new UnitOfWork($this);
```

Comment:
Why do you need a new instance of `UnitOfWork`? Why not let `clear` reset the rest of `UnitOfWork` state as well?

Reply:
I don't know, it was originally in d05e8e8